### DISCUSSION TITLE
USB drive emulation (for Wii games sharing)

### DISCUSSION THREAD

**Discussion by ilvise** [retronas/discussions#18] [Ideas]
Hi,
i just would put here an idea for having the Wii games on RetroNAS.
There is a lot of sbc that have the USB port configurable as usb DEVICE, such as the raspberry pi 0. Connecting a raspberry 0w to the wii via usb it can be possible to have alla the wii isos on retronas and via the wi-fi share that to the wii. The wii only see a normal usb drive. Is not that simple but is possible, afaik that involve the use of a disk image that is virtually the thumb drive. Having easy access to a disk image and the ability to modify that via a network share is not super easy but possible with some tricky-mount.
Of course we will need another raspberry for this, but it seems cool.
Here some example

https://web.archive.org/web/20200708130751/https://pi.gbaman.info/?p=699
https://raspberrypi.stackexchange.com/questions/67590/usb-mass-storage-gadget-how-to-convincingly-emulate-a-flash-drive
https://raspberrypi.stackexchange.com/questions/33545/rpi-to-emulate-a-usb-mass-storage-device

**Comment by sairuk** [comment]
We have support for setting up a RetroNAS device in mass_storage USB gadget mode and have it connect to an ISCSI target on another NAS for this kind of operation.

The main discussion is here https://github.com/danmons/retronas/discussions/287

Once the work on an image/iscsi/block manager is completed we can move forward with iscsi/images on RetroNAS

Our current doco is here: https://github.com/danmons/retronas/wiki/Gadgets
